backend/controllers/donation_controller.py

```
@@ -0,0 +1,588 @@
+            raise HTTPException(403, "You are not allowed to cancel this subscription")
+
+    # Call PayPal
+    payload = {"reason": "User requested cancellation"}
+    resp = await helper.post(
+        f"/v1/billing/subscriptions/{sub_id}/cancel",
+        json_body=payload,
+    )
+    if resp.status_code not in (200, 204):
+        try:
+            data = resp.json()
+        except Exception:
```

Ruff (0.14.5)

Do not catch blind exception: `Exception`

(BLE001)